### PR TITLE
PRIME-2028 Match PLR data better 

### PR DESCRIPTION
**Pull request by neophyte57**
| Scenario                                                                                                                                                                                                                       | Expected                                                                                                                                                              |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| 1\. Load \`PlrProvider\` table, e.g. by using PlrIntakeUtility                                                                                                                                                                 |  |
| 2\. Modify two \`PlrProvider\` records, with different role types belonging to two different colleges (e.g. PHARM and MD) so that both records have the same \`CollegeId\` value                                               |
| 3\. Enroll with a Certification matching one of the colleges (e.g. CPBC or CPSBC) and with license number matching PLR \`PlrProvider\` \`CollegeId\` value                                                                     |
| 4\. As Admin, review the enrollment in the Overview screen                                                                                                                                                                     | Only the details from the \`PlrProvider\` record with \`ProviderRoleType\` having the same college as the Enrollee certification should appear on the Overview screen
| 1\. Enroll with a certification with a license number matching a \`PlrProvider.CollegeId\` value but with a different college than the college associated with the \`PlrProvider.ProviderRoleType\`                            |  |
| 2\. As Admin, review the enrollment in the Overview screen                                                                                                                                                                     | No \`PlrProvider\` record details should show in Enrollee Overview                                                                                                   
| 1\. Modify two \`PlrProvider\` records, with different role types belonging to the same college (e.g. PO and MD) so that both records have the same \`CollegeId\` value. Assumption is that in reality this could never happen |                                      |
| 2\. Enroll with a Certification matching the college (e.g. CPSBC) and with license number matching PLR \`PlrProvider\` \`CollegeId\` value                                                                                     |
| 3\. As Admin, review the enrollment in the Overview screen                                                                                                                                                                     | No \`PlrProvider\` record details should show in Enrollee Overview and data error should be handled gracefully

### CODE REVIEW

File: prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public partial class AddCollegeForPlrRoleType : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "CollegeForPlrRoleType",
+                columns: table => new
+                {
+                    RoleTypeCode = table.Column<string>(type: "text", nullable: false),
+                    CollegeId = table.Column<int>(type: "integer", nullable: false)
```
[the remark this answers is not in the document]
Comment (neophyte57) [reply prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs:14]: EF Core defaults to `not null`, that is, `nullable: false`?  As implemented:  

image

**Review by johnnyl6 — APPROVED** [review]
LGTM

File: prime-dotnet-webapi/Migrations/20220111175626_AddCollegeForPlrRoleTypeAndPlrAdditions.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Prime.Migrations
+{
+    public partial class AddCollegeForPlrRoleTypeAndPlrAdditions : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "CollegeForPlrRoleType",
```
Comment (AMiKaH) [inline prime-dotnet-webapi/Migrations/20220111175626_AddCollegeForPlrRoleTypeAndPlrAdditions.cs:10]: We have two migrations adding this table causing error 'relation "CollegeForPlrRoleType" already exists'. You may need to add both migrations as one.

File: prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Prime.Migrations
+{
+    public partial class AddCollegeForPlrRoleType : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "CollegeForPlrRoleType",
```
Comment (AMiKaH) [inline prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs:10]: We have two migrations adding this table causing error 'relation "CollegeForPlrRoleType" already exists'. You may need to add both migrations as one.

Comment (AMiKaH) [inline prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs:10]: We have two migrations adding this table causing error 'relation "CollegeForPlrRoleType" already exists'. You may need to add both migrations as one.

**Review by AMiKaH — APPROVED** [review]

**Review by Paahn — APPROVED** [review]

File: prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs (hunk 3 above)
Comment (neophyte57) [reply prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs]: Removed earlier migration explicitly with `git rm prime-dotnet-webapi/Migrations/20220106014622_AddCollegeForPlrRoleType.cs` as it didn't show up in VS Code Git view or `git status`.  Now trying to refresh PR environment's database

**Pull request merged**